### PR TITLE
Cart Second Update

### PR DESCRIPTION
**Pull request by jackyw2018**
Updates:
1. Cart now can update quantity; the quantity persist after switching to other pages
2. The subtotal takes into quantity into account
3. The count in top right cart icon now counts the total quantity instead of the number of row of lineItems

### CODE REVIEW

File: src/reducers/lineItemReducer.js [hunk 1]
```diff
@@ -51,11 +51,11 @@ export const addLineItem = item => {
 
 // Updated line item quantity
 // After updating db will fetch all the line items for the cart
-export const updateLineItem = lineItem => {
+export const updateLineItem = (lineItemId, cartId, quantity) => {
   return dispatch => {
     return axios
-      .put(`api/lineItems/${lineItem.id}`, lineItem)
-      .then(() => dispatch(getLineItems(lineItem.cartId)));
+      .put(`api/lineItems/${lineItemId}`, quantity)
```
Comment (alexb823) [inline src/reducers/lineItemReducer.js]: Is this change necessary? This is going to brake handleAddToCart function in the ProductList and SingleProduct components.  That function passed the whole lineItem object to the updateLineItem function. Can't we just pass the whole object here, and then on the server side pass the whole lineItem obj to lineItem.update in the PUT route?
Any properties that changed will be updated.

Comment (jackyw2018) [reply src/reducers/lineItemReducer.js]: Yes, I thought of this for a bit too; I will later today push the revised code reserving the original thunk and make another PR
